iOS/DuckDuckGo/DuckPlayer/NativeUI/Views/DuckPlayerContainer.swift

```diff
@@ -122,55 +141,123 @@ private struct SheetView<Content: View>: View {
+    @GestureState private var dragStartOffset: Double?
+    @State private var isDragging = false
+    @State private var isAnimatingToTop = false
 
     // Animate the sheet offset with a spring animation
     private func animateOffset(to visible: Bool) {
-
+        let offset = calculateSheetOffset(for: visible, containerHeight: containerHeight)
         if #available(iOS 17.0, *) {
             withAnimation(
                 .spring(
-                    duration: DuckPlayerContainer.Constants.Animation.springDuration, bounce: DuckPlayerContainer.Constants.Animation.springBounce)
+                    duration: DuckPlayerContainer.Constants.springDuration, bounce: DuckPlayerContainer.Constants.springBounce)
             ) {
-                sheetOffset = calculateSheetOffset(for: visible, containerHeight: containerHeight)
+                sheetOffset = offset
             } completion: {
                 viewModel.sheetAnimationCompleted = true
             }
         } else {
             withAnimation(
                 .spring(
-                    duration: DuckPlayerContainer.Constants.Animation.springDuration, bounce: DuckPlayerContainer.Constants.Animation.springBounce)
+                    duration: DuckPlayerContainer.Constants.springDuration, bounce: DuckPlayerContainer.Constants.springBounce)
             ) {
-                sheetOffset = calculateSheetOffset(for: visible, containerHeight: containerHeight)
+                sheetOffset = offset
             }
-            // For earlier iOS versions, set completed immediately
             viewModel.sheetAnimationCompleted = true
         }
     }
 
     var body: some View {
-        VStack(alignment: .center) {
-
+        VStack(alignment: .center, spacing: 0) {
             if let sheetWidth {
-                content(DuckPlayerContainer.PresentationMetrics(contentWidth: sheetWidth))
+                VStack(spacing: 0) {
+                    ZStack(alignment: .top) {
+                        GrabHandle()
+
+                        content(DuckPlayerContainer.PresentationMetrics(contentWidth: sheetWidth))
+                            .padding(.top, DuckPlayerContainer.Constants.contentTopPadding)
+
+                        Rectangle()
+                            .fill(Color.clear)
+                            .frame(height: DuckPlayerContainer.Constants.dragAreaHeight)
+                            .contentShape(Rectangle())
+                            .gesture(
+                                DragGesture()
+                                    .updating($dragStartOffset) { _, state, _ in
+                                        if state == nil {
+                                            state = sheetOffset
+                                            viewModel.setDragging(true)
+                                        }
+                                    }
+                                    .onChanged { value in
```

Comment:
Maybe extract this
`private func handleDragGesture(value: DragGesture.Value, dragStartOffset: Double?) {`